apps/bilateral_grid/Makefile

```diff
@@ -1,22 +1,22 @@
 include ../support/Makefile.inc
 
-all: filter
+all: $(BIN)/filter
 
-bilateral_grid: bilateral_grid.cpp
-	$(CXX) $(CXXFLAGS) bilateral_grid.cpp -g $(LIB_HALIDE) -o bilateral_grid -lz $(LDFLAGS) \
-	$(LLVM_SHARED_LIBS)
+$(BIN)/bilateral_grid.a: bilateral_grid_generator.cpp
+	@-mkdir -p $(BIN)
+	$(HALIDE_SRC_PATH)/tools/gengen.sh -c $(CXX) -l $(LIB_HALIDE) -o $(BIN) -s $^ target=host
 
-bilateral_grid.a: bilateral_grid
-	./bilateral_grid 8
+$(BIN)/filter: $(BIN)/bilateral_grid.a filter.cpp
+	@-mkdir -p $(BIN)
+	$(CXX) $(CXXFLAGS) -O3 -ffast-math -Wall -Werror -I$(BIN) filter.cpp $(BIN)/bilateral_grid.a -o $@ $(PNGFLAGS) $(LDFLAGS)
 
-filter: bilateral_grid.a filter.cpp
-	$(CXX) $(CXXFLAGS) -O3 -ffast-math -Wall -Werror filter.cpp bilateral_grid.a -o filter  $(PNGFLAGS) $(LDFLAGS)
+$(BIN)/bilateral_grid.mp4: bilateral_grid_generator.cpp viz.sh
```

Comment:
I think this should depend on filter rather than bilateral_grid_generator.cpp. If filter.cpp changes then the mp4 will need to be remade. That's possibly why there used to be a make clean in the vis script.